crates/rust-analyzer/src/cli/load_cargo.rs

```diff
@@ -14,13 +14,22 @@ use vfs::{loader::Handle, AbsPath, AbsPathBuf};
 
 use crate::reload::{ProjectFolders, SourceRootConfig};
 
-pub(crate) struct LoadCargoConfig {
-    pub(crate) load_out_dirs_from_check: bool,
-    pub(crate) with_proc_macro: bool,
-    pub(crate) prefill_caches: bool,
+// Note: Since this type is used by external tools that use rust-analyzer as a library
+// what otherwise would be `pub(crate)` has to be `pub` here instead.
+#[allow(unreachable_pub)]
```

Comment:
is the `allow` still necessary with the `load_cargo` module being public?

Reply:
Probably not. I only noticed that the module had been made private _after_ adding them.

Reply:
Fixed.